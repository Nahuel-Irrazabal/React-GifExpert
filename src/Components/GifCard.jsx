import PropTypes from 'prop-types'

export const GifCard = ({title, url}) => {
  return (
    <div className="card">
      <h6>{title}</h6>
      <img src={ url } alt={ title }/>
    </div>
  )
}

/*  yarn add
  a√±adir proptypes
  titulo y url obligatorio

  evaluar un snapshot
  */

GifCard.protoTypes = {
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired
}