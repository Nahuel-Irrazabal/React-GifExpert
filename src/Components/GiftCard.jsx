
export const GiftCard = ({title, url}) => {
  return (
    <div className="card">
      <h6>{title}</h6>
      <img src={ url } alt={ title }/>
    </div>
  )
}
