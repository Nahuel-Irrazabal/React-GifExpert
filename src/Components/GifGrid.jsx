import { GifCard } from './GifCard'
import { useFetchGift } from './Hooks/useFetchGift'

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGift( category )

  return (
    <>
      <h3> {category} </h3>
      {
        isLoading && (<h5>Cargando...</h5>)
      }
      <div className='card-grid'>
        {
          images.map( ( image ) => (
            <GifCard 
              key={image.id}
              {...image}
            /> 
          ))
        }
      </div>
    </>
  )
}
