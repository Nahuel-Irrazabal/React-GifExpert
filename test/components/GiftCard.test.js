import { render } from "@testing-library/react"
import { GifCard } from "../../src/Components/GiftCard"

describe('Test GiftCard', () => {

    const title = 'Title Card'
    const url = 'url Card'

    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <GifCard title={title} url={url} /> )
        expect( container ).toMatchSnapshot()
    })

})