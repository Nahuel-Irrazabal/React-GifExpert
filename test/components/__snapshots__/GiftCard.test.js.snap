// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GiftCard debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h6>
      Title Card
    </h6>
    <img
      alt="Title Card"
      src="url Card"
    />
  </div>
</div>
`;
